cloudant-sync-datastore-core/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -117,7 +117,12 @@ public HttpConnection setNumberOfRetries(int numberOfRetries){
      * @return an {@link HttpConnection} for method chaining 
      */
     public HttpConnection setRequestBody(final String input) {
-        this.input = new ByteArrayInputStream(input.getBytes());
+        this.input = new InputStreamGenerator() {
+            @Override
+            public InputStream getInputStream() {
+                return new ByteArrayInputStream(input.getBytes());
```

Comment:
This uses the platforms default charset, I think we should probably use the more explicit `getBytes("UTF-8")`.